cdap-common/src/main/java/co/cask/cdap/common/lang/ProgramResources.java

```diff
@@ -115,8 +115,8 @@ public String apply(ClassPath.ClassInfo input) {
     Set<String> resources = getBaseResources();
 
     // Base on the type, add extra resources
-    // Current only Spark type has extra visible resources
-    if (type == ProgramType.SPARK) {
+    // Current only Spark and Workflow type has extra visible resources
+    if (type == ProgramType.SPARK || type == ProgramType.WORKFLOW) {
```

Comment:
Yes. I don't know how it works before. When running Spark inside Workflow, we use the same classloader as the Workflow program to load the Spark program.